.github/workflows/tester.yml

```diff
@@ -0,0 +1,73 @@
+name: "Testing Pull Request"
+
+on:
+  pull_request:
+    branches:
+      - "master"
+      - "dev"
+
+jobs:
+  unittests: #################################################################
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, macos-latest, ubuntu-latest]
+        python-version: [3.8, 3.9, '3.10', '3.11', '3.12']
+        
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install Python dependencies
+      run: |
+        python3 -m pip install --upgrade pip
+        python3 -m pip install .[test]
+
+    - name: Test with pytest
+      run: |
+        python3 -m pytest
+
+  linter: ####################################################################
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Run Black formatter (check mode)
+        uses: psf/black@stable
+        with:
```

Comment:
please add `options: "--check -l 80"` so it is compliant with `code_formatter.sh`

Reply:
already in pyproject.toml!

Reply:
Ok, then we need to install also dev dependencies in line 27 I think